internal/keppel/manifest.go

```diff
@@ -147,3 +160,19 @@ func (a listManifestAdapter) ManifestReferences(pf PlatformFilter) []manifestlis
 	}
 	return result
 }
+
+func (a listManifestAdapter) AcceptableAlternates(pf PlatformFilter) []manifestlist.ManifestDescriptor {
+	var result []manifestlist.ManifestDescriptor
+	for _, m := range a.ManifestReferences(pf) {
+		//If we have an application/vnd.docker.distribution.manifest.list.v2+json manifest, but the
+		//client only accepts application/vnd.docker.distribution.manifest.v2+json, in order to stay
```

Comment:
Do we know any client that does that? Why not check if the list contains only one entry and choose that?

Reply:
This is relevant for `docker pull`. The "only one entry" case does not apply because this is specifically about multi-arch images.